avro-fastserde/src/jmh/java/com/linkedin/avro/fastserde/FastAvroSerdesBenchmark.java

```diff
@@ -126,22 +127,48 @@ public void testFastAvroSerialization(Blackhole bh) throws Exception {
   @Benchmark
   @OperationsPerInvocation(NUMBER_OF_OPERATIONS)
   public void testAvroDeserialization(Blackhole bh) throws Exception {
-    testDeserialization(deserializer, bh);
+    testDeserialization(deserializer, bh, 0);
   }
 
   @Benchmark
   @OperationsPerInvocation(NUMBER_OF_OPERATIONS)
   public void testFastAvroDeserialization(Blackhole bh) throws Exception {
-    testDeserialization(fastDeserializer, bh);
+    testDeserialization(fastDeserializer, bh, 0);
   }
 
-  private void testDeserialization(DatumReader<GenericRecord> datumReader, Blackhole bh) throws Exception {
+  @Benchmark
+  @OperationsPerInvocation(NUMBER_OF_OPERATIONS)
+  public void testFastAvroDeserializationAccess(Blackhole bh) throws Exception {
+    testDeserialization(fastDeserializer, bh, 1);
+  }
+
+  @Benchmark
+  @OperationsPerInvocation(NUMBER_OF_OPERATIONS)
+  public void testFastAvroDeserializationAccess8(Blackhole bh) throws Exception {
+    testDeserialization(fastDeserializer, bh, 8);
+  }
+
+  @Benchmark
+  @OperationsPerInvocation(NUMBER_OF_OPERATIONS)
+  public void testFastAvroDeserializationAccess16(Blackhole bh) throws Exception {
+    testDeserialization(fastDeserializer, bh, 16);
+  }
+
+  private void testDeserialization(DatumReader<GenericRecord> datumReader, Blackhole bh, int access) throws Exception {
```

Comment:
nit - i'd rename "access" to "numElementsToAccess" to make it slightly clearer? (or add javadoc)

Reply:
Sure, will rename to make it more meaninful